src/pages/studyView/StudyViewUtils.tsx

```diff
@@ -2798,3 +2798,13 @@ export function getMolecularProfileSamplesSet(
         {}
     );
 }
+
+export function conditionalInvocation<T>(
```

Comment:
not really what i meant.   this is just a replacement for "if" which does not improve on it.  also, you have a mystery return type [T] which isn't suggested by the name.  
what i mean is  instead of this: 
```this.studyViewFilterWithFilteredSampleIdentifiers.result!.sampleIdentifiers.length > 0```
this:
```this.hasSamplesInCurrentFilter ```
or some name which spells out the condition.

Reply:
the name should suggest why we don't have to fetch anything